clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -2295,6 +2292,23 @@ class UnsafeBufferUsageReporter : public UnsafeBufferUsageHandler {
     }
   }
 
+  void handleUnsafeOperationInContainer(const Stmt *Operation,
+                                        bool IsRelatedToDecl,
+                                        ASTContext &Ctx) override {
+    SourceLocation Loc;
+    SourceRange Range;
+    unsigned MsgParam = 0;
+    if (const auto *CtorExpr = dyn_cast<CXXConstructExpr>(Operation)) {
```

Comment:
This should be a hard `cast<>()` because you don't really handle the else-branch so it results in an invalid source location. We can make it dynamic again later if we want.

Reply:
Got it. The old code was dyn but that was because it handled other cases too. Once this function handles other gadgets we'll adjust it.